<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1110</width>
    <height>562</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DES&amp;AES加解密程序 By 孙鹏-U201417253</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>1061</width>
      <height>521</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>DES加解密</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget_2">
      <property name="geometry">
       <rect>
        <x>920</x>
        <y>-10</y>
        <width>77</width>
        <height>331</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QPushButton" name="F_Btn">
         <property name="text">
          <string>计算</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="E_Btn">
         <property name="text">
          <string>计算</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="XOR_Btn">
         <property name="text">
          <string>计算</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="IP_Btn">
         <property name="text">
          <string>计算</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="Line" name="line_2">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>312</y>
        <width>571</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QWidget" name="gridLayoutWidget_2">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>332</y>
        <width>521</width>
        <height>151</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="verticalSpacing">
        <number>6</number>
       </property>
       <item row="0" column="3">
        <widget class="QPushButton" name="cipher_btn_2">
         <property name="text">
          <string>加密</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>明文文件路径</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>密文文件路径</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QPushButton" name="decipher_btn_2">
         <property name="text">
          <string>解密</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>    密钥</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="ciphered_text_2">
         <property name="maxLength">
          <number>30000</number>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="uncipher_text_2">
         <property name="maxLength">
          <number>30000</number>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLabel" name="tip_label_2">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="file_select_btn">
         <property name="text">
          <string>选择文件</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="key_2">
         <property name="maxLength">
          <number>16</number>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="file_select_btn_2">
         <property name="text">
          <string>选择文件</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="gridLayoutWidget">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>12</y>
        <width>451</width>
        <height>131</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_13">
       <property name="verticalSpacing">
        <number>6</number>
       </property>
       <item row="3" column="1">
        <widget class="QLabel" name="tip_label">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QPushButton" name="cipher_btn">
         <property name="text">
          <string>加密</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>密文</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>明文</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QPushButton" name="decipher_btn">
         <property name="text">
          <string>解密</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>密钥</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="ciphered_text">
         <property name="maxLength">
          <number>16</number>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="key">
         <property name="maxLength">
          <number>16</number>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="uncipher_text">
         <property name="maxLength">
          <number>16</number>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>中间过程：</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>-18</y>
        <width>20</width>
        <height>501</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
     <widget class="QWidget" name="verticalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>12</y>
        <width>401</width>
        <height>288</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>F(R,K)计算</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="1">
          <widget class="QLineEdit" name="F_K_Input">
           <property name="maxLength">
            <number>12</number>
           </property>
           <property name="placeholderText">
            <string>输入K</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLineEdit" name="F_R_Input">
           <property name="maxLength">
            <number>8</number>
           </property>
           <property name="placeholderText">
            <string>输入R</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLineEdit" name="F_Output">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>E(R)计算</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="E_Input">
         <property name="maxLength">
          <number>8</number>
         </property>
         <property name="placeholderText">
          <string>输入R</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="E_Output">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>XOR计算</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="1">
          <widget class="QLineEdit" name="XOR_R">
           <property name="maxLength">
            <number>8</number>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLineEdit" name="XOR_L">
           <property name="maxLength">
            <number>8</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLineEdit" name="XOR_Output">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>IP计算</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="IP_Input">
         <property name="maxLength">
          <number>16</number>
         </property>
         <property name="placeholderText">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="IP_Output">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>142</y>
        <width>451</width>
        <height>321</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QTextBrowser" name="K_info"/>
       </item>
       <item>
        <widget class="QTextBrowser" name="Mid_info"/>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>AES加解密</string>
     </attribute>
     <widget class="QWidget" name="gridLayoutWidget_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>541</width>
        <height>131</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_14">
       <property name="verticalSpacing">
        <number>6</number>
       </property>
       <item row="3" column="1">
        <widget class="QLabel" name="tip_label_aes">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QPushButton" name="cipher_btn_aes">
         <property name="text">
          <string>加密</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_34">
         <property name="text">
          <string>密文</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_35">
         <property name="text">
          <string>明文</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QPushButton" name="decipher_btn_aes">
         <property name="text">
          <string>解密</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_36">
         <property name="text">
          <string>密钥</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="ciphered_text_aes">
         <property name="maxLength">
          <number>32</number>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="key_aes">
         <property name="maxLength">
          <number>32</number>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="uncipher_text_aes">
         <property name="maxLength">
          <number>32</number>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLabel" name="label_37">
         <property name="text">
          <string>中间过程：</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>150</y>
        <width>541</width>
        <height>321</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="10,3">
       <property name="spacing">
        <number>6</number>
       </property>
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QTextBrowser" name="Mid_info_aes"/>
       </item>
       <item>
        <widget class="QTextBrowser" name="K_info_aes"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="gridLayoutWidget_4">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>10</y>
        <width>461</width>
        <height>348</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_16">
       <item row="5" column="0">
        <widget class="QLineEdit" name="shiftrows_output">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLineEdit" name="mixcol_input">
         <property name="maxLength">
          <number>32</number>
         </property>
         <property name="placeholderText">
          <string>输入state</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QPushButton" name="sr_btn">
         <property name="text">
          <string>计算</string>
         </property>
        </widget>
       </item>
       <item row="9" column="0">
        <widget class="QLabel" name="label_41">
         <property name="text">
          <string>AddRoundKey计算</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QPushButton" name="mc_btn">
         <property name="text">
          <string>计算</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_40">
         <property name="text">
          <string>MixColumns计算</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLineEdit" name="shiftrows_input">
         <property name="maxLength">
          <number>32</number>
         </property>
         <property name="placeholderText">
          <string>输入state</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLineEdit" name="subbytes_output">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="sb_btn">
         <property name="text">
          <string>计算</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QLineEdit" name="mixcol_output">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="13" column="0">
        <widget class="QLineEdit" name="addroundkey_output">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_39">
         <property name="text">
          <string>ShiftRows计算</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_38">
         <property name="text">
          <string>SubBytes计算</string>
         </property>
        </widget>
       </item>
       <item row="11" column="1">
        <widget class="QPushButton" name="ark_btn">
         <property name="text">
          <string>计算</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLineEdit" name="subbytes_input">
         <property name="text">
          <string/>
         </property>
         <property name="maxLength">
          <number>32</number>
         </property>
         <property name="placeholderText">
          <string>输入state</string>
         </property>
        </widget>
       </item>
       <item row="11" column="0">
        <widget class="QLineEdit" name="addroundkey_input">
         <property name="maxLength">
          <number>32</number>
         </property>
         <property name="placeholderText">
          <string>输入state</string>
         </property>
        </widget>
       </item>
       <item row="12" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="5,1">
         <item>
          <widget class="QLineEdit" name="addroundkey_key_input">
           <property name="maxLength">
            <number>32</number>
           </property>
           <property name="placeholderText">
            <string>输入key</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="addroundkey_round_input">
           <property name="maxLength">
            <number>2</number>
           </property>
           <property name="placeholderText">
            <string>输入Round</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="gridLayoutWidget_5">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>370</y>
        <width>461</width>
        <height>121</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_17">
       <property name="verticalSpacing">
        <number>6</number>
       </property>
       <item row="0" column="3">
        <widget class="QPushButton" name="cipher_file_btn_aes">
         <property name="text">
          <string>加密</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_42">
         <property name="text">
          <string>明文文件路径</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_43">
         <property name="text">
          <string>密文文件路径</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QPushButton" name="decipher_file_btn_aes">
         <property name="text">
          <string>解密</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_44">
         <property name="text">
          <string>    密钥</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="ciphered_file_path">
         <property name="maxLength">
          <number>30000</number>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="uncipher_file_path">
         <property name="maxLength">
          <number>30000</number>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLabel" name="tip_file_label_aes">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="uncipher_file_select_btn_aes">
         <property name="text">
          <string>选择文件</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="key_file">
         <property name="maxLength">
          <number>32</number>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="ciphered_file_select_btn_aes">
         <property name="text">
          <string>选择文件</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="Line" name="line_3">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>359</y>
        <width>511</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="Line" name="line_4">
      <property name="geometry">
       <rect>
        <x>543</x>
        <y>-30</y>
        <width>31</width>
        <height>541</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
